core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -172,10 +172,10 @@ class TaskMetrics private[spark](
     val resultSerializationTime: Long,
     val memoryBytesSpilled: Long,
     val diskBytesSpilled: Long,
-    val inputMetrics: Option[InputMetrics],
-    val outputMetrics: Option[OutputMetrics],
-    val shuffleReadMetrics: Option[ShuffleReadMetrics],
-    val shuffleWriteMetrics: Option[ShuffleWriteMetrics])
+    val inputMetrics: InputMetrics,
```

Comment:
I couldn't find the jackson doc for it, so I tried it locally. For scala `Option`, jackson just treat it as a nullable field, if it's `Some`, it will unwrap the `Some` and parse the real value. If it's `None`, it will return null for that field.

In this case, where we always return `Some`, it's ok to drop the `Option`.